browser/profiles/brave_profile_manager_browsertest.cc

```diff
@@ -118,3 +177,173 @@ IN_PROC_BROWSER_TEST_F(BraveProfileManagerTest, MigrateProfileNames) {
+  ASSERT_TRUE(brave::IsTorProfile(tor_profile));
+  EXPECT_TRUE(tor_profile->IsOffTheRecord());
+  EXPECT_EQ(tor_profile->GetParentProfile(), parent_profile);
+
+  // Check Tor profile's content settings are inherited from its parent.
+  HostContentSettingsMap* tor_content_settings =
+    HostContentSettingsMapFactory::GetForProfile(tor_profile);
+  ContentSetting setting = GetScriptSetting(tor_content_settings, brave_url);
+  EXPECT_EQ(setting, CONTENT_SETTING_BLOCK);
+
+  // Check changes of content settings from the parent profile will reflected
+  // in Tor profile when the setting is not set directly in Tor profile.
+  SetScriptSetting(parent_content_settings,
+      ContentSettingsPattern::Wildcard(), CONTENT_SETTING_ALLOW);
+  setting = GetScriptSetting(tor_content_settings, brave_url);
+  EXPECT_EQ(setting, CONTENT_SETTING_ALLOW);
+
+  // Check changes of content settings from the parent profile will not
+  // overwrite the one in Tor profile when it is set directly in Tor profile.
+  SetScriptSetting(tor_content_settings,
+      ContentSettingsPattern::FromURL(brave_url),
+      CONTENT_SETTING_BLOCK);
+  setting = GetScriptSetting(parent_content_settings, brave_url);
+  EXPECT_EQ(setting, CONTENT_SETTING_ALLOW);
+  setting = GetScriptSetting(tor_content_settings, brave_url);
+  EXPECT_EQ(setting, CONTENT_SETTING_BLOCK);
+}
+
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+class BraveProfileManagerExtensionTest
+  : public extensions::ExtensionBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    extensions::ExtensionBrowserTest::SetUpOnMainThread();
+
+    // Override extension data dir.
+    brave::RegisterPathProvider();
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir_);
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(BraveProfileManagerExtensionTest,
+    SwitchToTorProfileShareExtensions) {
+  ScopedTorLaunchPreventerForTest prevent_tor_process;
+  ProfileManager* profile_manager = g_browser_process->profile_manager();
+  ASSERT_TRUE(profile_manager);
+  Profile* parent_profile = ProfileManager::GetActiveUserProfile();
+  ASSERT_TRUE(parent_profile);
+
+  // Install an extension in parent profile.
+  const extensions::Extension* extension = InstallExtension(
+      test_data_dir_.AppendASCII("trivial_extension"), 1);
+  const std::string id = extension->id();
+
+  Profile* tor_profile = SwitchToTorProfile();
+  ASSERT_TRUE(brave::IsTorProfile(tor_profile));
+  EXPECT_TRUE(tor_profile->IsOffTheRecord());
+  EXPECT_EQ(tor_profile->GetParentProfile(), parent_profile);
+
+  // Extension should be disabled in Tor by default.
+  EXPECT_FALSE(extensions::util::IsIncognitoEnabled(id, tor_profile));
+
+  // Change of enabled in incognito should be reflected immediately in Tor.
+  extensions::ExtensionPrefs* parent_extension_prefs =
+      extensions::ExtensionPrefs::Get(parent_profile);
+  parent_extension_prefs->SetIsIncognitoEnabled(id, true);
+  EXPECT_TRUE(extensions::util::IsIncognitoEnabled(id, tor_profile));
```

Comment:
can we also check `ExtensionRegistry` here to make sure tor profile have the extension installed?

Reply:
Addressed in https://github.com/brave/brave-core/pull/3319/commits/e791d30dc1f34e600c430d7b427fe6aae2599b55